### PR TITLE
Use Kubernetes 1.15.4 and architect-orb 0.4.2

### PR DESCRIPTION
**Pull request by rossf7**
Towards https://github.com/giantswarm/giantswarm/issues/6302

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 version: 2.1
 orbs:
-  architect: giantswarm/architect@0.2.0
+  architect: giantswarm/architect@0.4.1
```
Comment (rossf7) [inline .circleci/config.yml]: This was outdated. Everything else is dep stuff.

Comment (kopiczko) [inline .circleci/config.yml]:
```suggestion
  architect: giantswarm/architect@0.4.2
```

Comment (kopiczko) [reply .circleci/config.yml]: Fresh from the oven.

Comment (rossf7) [reply .circleci/config.yml]: Nice! 🍞 All sorted :)

**Pull request merged**